plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -188,9 +188,30 @@ void GSSettingsDlg::OnInit()
 	AddTooltip(IDC_LOGZ);
 	AddTooltip(IDC_LARGE_FB);
 
+	UpdateFilteringCombobox();
 	UpdateControls();
 }
 
+void GSSettingsDlg::UpdateFilteringCombobox()
+{
+	INT_PTR i;
+	ComboBoxGetSelData(IDC_RENDERER, i);
+	bool OpenGL = static_cast<GSRendererType>(i) == GSRendererType::OGL_HW;
+	uint8 filter = (ComboBoxGetSelData(IDC_FILTER, i)) ? static_cast<uint8>(i) : static_cast<uint8>(theApp.GetConfigI("filter"));
+
+	vector<GSSetting> list = theApp.m_gs_filter;
+	if (!OpenGL) //Currently Trilinear is only exclusive to OpenGL, remove those combobox items when any other renderer is used
+	{
+		for (uint8 option = uint8(Filtering::Trilinear_Always); option != uint8(Filtering::Bilinear_PS2); option--)
+			list.erase(list.begin() + option);
```

Comment:
`erase()` can remove multiple items, so the `for` loop is unnecessary.

Alternatively, instead of copying `theApp.m_gs_filter` and then erasing the trilinear options, you could simply copy only the necessary vector elements using `insert()`.